src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1486,15 +1512,11 @@ private String launchCommandWithCredentials(ArgumentListBuilder args, File workD
         }
     }
 
-    private File createSshKeyFile(File key, SSHUserPrivateKey sshUser) throws IOException, InterruptedException {
-        key = File.createTempFile("ssh", "key");
-        try (PrintWriter w = new PrintWriter(key)) {
-            List<String> privateKeys = sshUser.getPrivateKeys();
-            for (String s : privateKeys) {
-                w.println(s);
-            }
-        }
-        new FilePath(key).chmod(0400);
+    private FilePath createSshKeyFile(SSHUserPrivateKey sshUser) throws IOException, InterruptedException {
+        FilePath key = workspace.createTempFile("ssh", "key");
+        List<String> privateKeys = sshUser.getPrivateKeys();
+        key.write(StringUtils.join(privateKeys, "\n"), null);
```

Comment:
smells like a bug to me the OS may be windows which does not use '\n' but '\r\n'